### ISSUE TITLE
Eas build fails when expo-updates is installed

### ISSUE THREAD

**Issue by IggsGrey** [expo/expo#19112]
### Summary

After upgrading my expo project to use SDK 45, running eas build fails with the error: "Project with path ':expo-updates-interface' could not be found in project ':expo-updates'"

Of course it works without expo-updates installed.

I installed expo-updates with expo install

### What platform(s) does this occur on?

Not Applicable

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: Linux 5.13 Ubuntu 21.10 21.10 (Impish Indri)
      Shell: 5.1.8 - /bin/bash
    Binaries:
      Node: 16.6.1 - ~/.nvm/versions/node/v16.6.1/bin/node
      npm: 7.20.3 - ~/.nvm/versions/node/v16.6.1/bin/npm
    SDKs:
      Android SDK:
        API Levels: 29, 30
        Build Tools: 29.0.2, 30.0.2, 30.0.3
        System Images: android-30 | Google APIs Intel x86 Atom
    npmPackages:
      expo: ^45.0.0 => 45.0.8 
      react: 17.0.2 => 17.0.2 
      react-dom: 17.0.2 => 17.0.2 
      react-native: 0.68.2 => 0.68.2 
      react-native-web: 0.17.7 => 0.17.7 
    npmGlobalPackages:
      eas-cli: 0.55.1
      expo-cli: 6.0.5
    Expo Workflow: managed

### Minimal reproducible example

https://github.com/IggsGrey/Eas-build-error-reproduction

**Comment by IggsGrey** [comment]
#18750

**Comment by brentvatne** [comment]
can you link to a build page on eas build that failed?

**Comment by IggsGrey** [comment]
https://expo.dev/accounts/iggs_grey/projects/wambo/builds/b6be7240-84e0-40df-a9dc-c12e105afcda

**Comment by IggsGrey** [comment]
> https://expo.dev/accounts/iggs_grey/projects/wambo/builds/b6be7240-84e0-40df-a9dc-c12e105afcda

@brentvatne if this satisfies your request, could you remove the `needs more info` label

**Comment by brentvatne** [comment]
it looks like you are using a monorepo and the dependency isn't found where expected. please refer to:

- https://docs.expo.dev/guides/monorepos/
- https://docs.expo.dev/build-reference/troubleshooting/ (i recommend trying to get the build run locally as suggested here)

**Comment by IggsGrey** [comment]
I'm not too sure what makes the reproducible example a monorepo but how can i modify it into a simple app without monorepos?

I tried the the local build in the troubleshooting guide earlier, but it failed with an error i was unsure of how to solve

![image](https://user-images.githubusercontent.com/41928390/190624142-84dd67f0-962d-44da-9e4d-c6ebcb1748a4.png)

**Comment by brentvatne** [comment]
@IggsGrey - the build that you linked me to is different from the example you shared. the build you linked me to is a monorepo

**Comment by IggsGrey** [comment]
@brentvatne
I am actually not paying much attention to the local build because i would like to avoid falling back to local builds (and i believe the issue may have to do with more than the build environment).

I would be happy to share with you a new repo and build page if that will be in place here (as this issue has been closed already, otherwise let me know if it will be more appropriate to create a new issue)

[Failed Eas Build](https://expo.dev/accounts/iggs_grey/projects/my-app/builds/ae079ded-20a0-4b03-9f60-b09ce6810096)
[Minimum reproducible example](https://github.com/IggsGrey/Eas-build-error-reproduction) : I created a new expo45 project with create-expo-app and followed the eas update getting started guide [here](https://docs.expo.dev/eas-update/getting-started/#create-a-build-for-the-project) to setup the project for updates.

I may have forgotten to add earlier that my project was initially using expo[42 and 44] all of which had successful builds with expo-updates installed, but after upgrading to expo45 it fails. #18750 also has this same issue with expo45.

Following the same steps here in an expo44 project builds successfully with eas build ([build page](https://expo.dev/accounts/iggs_grey/projects/your-app/builds/219f7dfc-f07c-4c1a-b211-c4ddebaf57a0))
